src/coprocessor/dag/expr/builtin_cast.rs

```diff
@@ -359,4 +456,82 @@ impl FnCall {
     ) -> Result<Option<Vec<Duration>>> {
         unimplemented!()
     }
+
+    pub fn cast_json_as_duration(
+        &self,
+        ctx: &StatementContext,
+        row: &[Datum],
+    ) -> Result<Option<Vec<Duration>>> {
+        unimplemented!()
+    }
+
+    pub fn cast_int_as_json(
+        &self,
+        ctx: &StatementContext,
+        row: &[Datum],
+    ) -> Result<Option<Vec<Json>>> {
+        unimplemented!()
+    }
+
+    pub fn cast_real_as_json(
+        &self,
+        ctx: &StatementContext,
+        row: &[Datum],
+    ) -> Result<Option<Vec<Json>>> {
+        unimplemented!()
+    }
+
+    pub fn cast_decimal_as_json(
+        &self,
+        ctx: &StatementContext,
+        row: &[Datum],
+    ) -> Result<Option<Vec<Json>>> {
+        unimplemented!()
+    }
+
+    pub fn cast_str_as_json(
+        &self,
+        ctx: &StatementContext,
+        row: &[Datum],
+    ) -> Result<Option<Vec<Json>>> {
+        unimplemented!()
+    }
+
+    pub fn cast_time_as_json(
+        &self,
+        ctx: &StatementContext,
+        row: &[Datum],
+    ) -> Result<Option<Vec<Json>>> {
+        unimplemented!()
+    }
+
+    pub fn cast_duration_as_json(
+        &self,
+        ctx: &StatementContext,
+        row: &[Datum],
+    ) -> Result<Option<Vec<Json>>> {
+        unimplemented!()
+    }
+
+    pub fn cast_json_as_json(
+        &self,
+        ctx: &StatementContext,
+        row: &[Datum],
+    ) -> Result<Option<Vec<Json>>> {
+        unimplemented!()
+    }
+
+    /// `produce_float_with_specified_tp`(`ProduceFloatWithSpecifiedTp` in tidb) produces
+    /// a new float64 according to `flen` and `decimal` in `self.tp`.
+    /// TODO port tests from tidb(tidb haven't implemented now)
+    fn produce_float_with_specified_tp(&self, ctx: &StatementContext, f: f64) -> Result<f64> {
+        let flen = self.tp.get_flen();
+        let decimal = self.tp.get_decimal();
+        if flen == convert::UNSPECIFIED_LENGTH || decimal == convert::UNSPECIFIED_LENGTH {
+            return Ok(f);
+        }
+        let dec = try!(Decimal::from_f64(f));
```

Comment:
I think reimplement `convert_to` for float is much cheaper than using decimal here.